### PR TITLE
Issue 25 - Redux User State

### PR DESCRIPTION
**Pull request by vincentchiang1**
### What problem are you solving in this PR?
*State the problem*
Store user data in redux

### Related issues/PRs:
*Resolves [https://github.com/siddhero97/CS436I-Memofy/issues/25]*

### How is this accomplished and what led to this approach?
*Explain what you implemented/what code you wrote*
Created actions, types, reducers, and selectors for user login and logout.

### What are the risks in this PR?
not sure how to test

*Does your change affect anything else in the app?*

### Before-I-merge checklist
- [] I have [self-tophatted]

### CODE REVIEW

**Comment by siddhero97** [comment]
Bro Fix the linter

**Comment by siddhero97** [comment]
I believe you should dispatch the login method when login is called and dispatch logout when logout is called from the login and Navbar.

**Comment by siddhero97** [comment]
How are you planning to store the password?

File: client/src/store/types.ts [hunk 1]
```diff
@@ -2,8 +2,16 @@ import {Item} from "./item/types";
...
 
 export interface ItemState {
   items: Item[];
+};
+
+export interface UserState {
+  name: string;
+  email: string;
+  phone: string;
+  isPremium: boolean;
```
Comment (KwanLam17) [inline client/src/store/types.ts:16]: Lets not add this yet.

File: client/src/store/user/reducers.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    isPremium: true
+  };
+
+export function userReducer(
+  state = initialState,
+  action: UserActionTypes,
+): UserState {
+  switch (action.type) {
+    case LOGIN:
+      return {
```
Comment (KwanLam17) [inline client/src/store/user/reducers.ts:18]: I think you can do `action.payload`?

File: client/src/store/user/reducers.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        name: action.payload.name,
+        email: action.payload.email,
+        phone: action.payload.phone,
+        isPremium: action.payload.isPremium
+      };
+    case LOGOUT:
+      return {
```
Comment (KwanLam17) [inline client/src/store/user/reducers.ts:25]: Can we set this to `null`?

**Pull request merged**